src/features/vote/SubmitModal.tsx

```diff
@@ -0,0 +1,156 @@
+import {
+  ForwardRefRenderFunction,
+  PropsWithChildren,
+  forwardRef,
+  Dispatch,
+  SetStateAction,
+  BaseSyntheticEvent,
+} from "react";
+
+import Modal from "common/components/modal";
+import Button from "common/components/button";
+import { ModalWrapper } from "./styled/SubmitModal.styled";
+import { SubmitModalState, Summary, FormData } from "./CommitPhase";
+import { UnlockedIcon, LockedIconCommitted } from "assets/icons";
+import {
+  FieldValues,
+  SubmitHandler,
+  SubmitErrorHandler,
+} from "react-hook-form";
+
+interface Props {
+  isOpen: boolean;
+  close: () => void;
+  ref: (node: HTMLElement | null) => void;
+  modalState: SubmitModalState;
+  setModalState: Dispatch<SetStateAction<SubmitModalState>>;
+  showModalSummary: () => Summary[];
+  // From react-hook-form
+  reset: (
+    values?:
+      | {
+          [x: string]: string | undefined;
+        }
+      | undefined,
+    omitResetState?:
+      | Partial<{
+          errors: boolean;
+          isDirty: boolean;
```

Comment:
What does `isDirty` does? It doesn't seem used anywhere

Reply:
Ripped from the lib's typescript. No idea.